oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/config/ProviderSettings.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.server.authorization.config;
+
+
+import org.springframework.util.Assert;
+
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * A facility for OpenID Connect Provider Configuration settings.
+ *
+ * @author Daniel Garnier-Moiroux
+ * @since 0.1.0
+ * @see Settings
+ * @see <a target="_blank" href="https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata">OpenID Connect Discovery 1.0</a>
+ */
+public class ProviderSettings extends Settings {
+	private static final String PROVIDER_SETTING_BASE = "spring.security.oauth2.authorization-server.provider.";
+	public static final String ISSUER = PROVIDER_SETTING_BASE.concat("issuer");
+	public static final String OPENID_CONFIGURATION_ENDPOINT = PROVIDER_SETTING_BASE.concat("openid-configuration-endpoint");
+	public static final String AUTHORIZATION_ENDPOINT = PROVIDER_SETTING_BASE.concat("authorization-endpoint");
+	public static final String TOKEN_ENDPOINT = PROVIDER_SETTING_BASE.concat("token-endpoint");
+	public static final String JWK_SET_ENDPOINT = PROVIDER_SETTING_BASE.concat("jwk-set-endpoint");
+	public static final String TOKEN_REVOCATION_ENDPOINT = PROVIDER_SETTING_BASE.concat("token-revocation-endpoint");
+
+	/**
+	 * Constructs a {@code ProviderSettings} from an Issuer URL.
+	 * The Issuer URL MUST also be a valid URI.
+	 *
+	 * @param issuer The URI representing the Issuer
+	 */
+	public ProviderSettings(URL issuer) {
+		super(defaultSettings());
+		Assert.notNull(issuer, "issuer cannot be null");
+		try {
+			issuer.toURI();
+		} catch (URISyntaxException e) {
+			throw new IllegalArgumentException("issuer must be a valid URI");
+		}
+		setting(ISSUER, issuer);
+	}
+
+	/**
+	 * Returns the URL for the OpenID Issuer.
+	 *
+	 * @return the URL for the OpenID Issuer
+	 */
+	public URL issuer() {
+		return setting(ISSUER);
+	}
+
+	/**
+	 * Returns the OpenID Provider Configuration endpoint. The default is {@code /.well-known/openid-configuration}.
+	 *
+	 * @return the OpenID Provider Configuration endpoint
+	 */
+	public String openIDProviderConfigurationEndpoint() {
```

Comment:
Let's remove this as the spec states it should be `/.well-known/openid-configuration`. I'm not sure if we need to allow this endpoint to be customized.